pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1576,6 +1589,24 @@ async def run(self):
                 ):
                     for d_content in d_content_by_nat_key[result.natural_key()]:
                         # ============ The below lines are added vs. pulpcore ============
+                        if model_type == Modulemd:
+                            # Fix saved snippet if malformed in DB, covers #2735
+                            natural_keys = d_content.content.natural_key_dict()
+                            defaults_q = Modulemd.objects.filter(**natural_keys)
```

Comment:
Why perform a new query here?  We already have both the incoming (`d_content.content`) and existing (`result`) model objects